compiler/rustc_hir_analysis/src/hir_ty_lowering/generics.rs

```diff
@@ -474,16 +475,9 @@ pub(crate) fn check_generic_arg_count(
             return Ok(());
         }
 
-        if provided_args > max_expected_args {
-            invalid_args.extend(
-                gen_args.args[max_expected_args..provided_args].iter().map(|arg| arg.span()),
-            );
-        };
+        invalid_args.extend(min_expected_args..provided_args);
```

Comment:
Will this not give a backwards range (i.e. `2..1`) if you have a `fn foo<'a: 'a, 'b: 'b>` and you call it `foo::<'static>`?

Reply:
Yes, but that's the prev behaviour. I just merged the two loops